app/src/processing/app/Editor.java

```diff
@@ -670,7 +673,15 @@ private void buildSketchMenu(JMenu sketchMenu) {
     item.addActionListener(event -> handleExport(false));
     sketchMenu.add(item);
 
-    item = newJMenuItemShift(tr("Upload Using Programmer"), 'U');
+//  Since CTRL+SHIFT+U is not working on iBus keyboard input method
+//  Lets redirect the shorcut for Linux to CTRL+ALT+U
+//  Leaving the preexisting behaviour for Windows & Mac OS
+    String OS = System.getProperty("os.name").toLowerCase();
```

Comment:
We have helpers to recognize operating systems, it's better to use them.
THis could become 
```
if (OSUtils.isWindows()) {
  item = newJMenuItemShift(tr("Upload Using Programmer"), 'U');
} else {
  item = newJMenuItemAlt(tr("Upload Using Programmer"), 'U');
}
```

Reply:
Thank you, I did not know that. 
I will commit the change with Arduino Helpers. 
But, I have to clarify that Mac OS should behave as WIndows, not as Linux. This is the case right now, and I do have concerns changing it.